homeassistant/components/vesync/__init__.py

```diff
@@ -135,7 +135,18 @@ async def async_new_device_discovery(service: ServiceCall) -> None:
 
 async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     """Unload a config entry."""
-    unload_ok = await hass.config_entries.async_unload_platforms(entry, PLATFORMS)
+    in_use_platforms = []
+    if hass.data[DOMAIN][VS_SWITCHES]:
+        in_use_platforms.append(Platform.SWITCH)
+    if hass.data[DOMAIN][VS_FANS]:
+        in_use_platforms.append(Platform.FAN)
+    if hass.data[DOMAIN][VS_LIGHTS]:
+        in_use_platforms.append(Platform.LIGHT)
+    if hass.data[DOMAIN][VS_SENSORS]:
+        in_use_platforms.append(Platform.SENSOR)
```

Comment:
For a future iteration, I would strong suggest to actually just move the detection on what to add to the platforms, this way you can just load all platforms, and then if a platform is not relevant, you can just not add entities